pymc_marketing/mmm/models/components/lagging.py

```diff
@@ -0,0 +1,394 @@
+from typing import Dict, Optional, Union
+
+import pymc as pm
+
+from pymc_marketing.mmm.transformers import geometric_adstock, weibull_adstock
+from pymc_marketing.mmm.utils import _get_distribution
+
+
+class BaseFunction:
```

Comment:
Think it would be good to setup the requirements of this class here. 
For instance, an ABC can be used and an `abstractmethod` for default_model_config

Also, using this for `apply` method would catch at class initialization instead of method call